src/arrayMethodJoin.js

```diff
@@ -5,8 +5,34 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    let str = '';
+    const last = this.length - 1;
+
+    let newSeparator = separator;
+
+    if (separator === undefined) {
+      newSeparator = ',';
+    }
+
+    if (separator === null) {
+      newSeparator = 'null';
+    }
+
+    for (let i = 0; i < this.length; i++) {
+      if (this[i] === undefined || this[i] === null) {
+        this[i] = '';
+      }
+      str += this[i];
+
+      if (i === last) {
+        break;
+      }
+
+      str += newSeparator;
```

Comment:
Here you can modify your condition to: `if i is not last - add separator`. Code will be shorter, cause you wouldn't need `break`.

Reply:
Great! Thank you!!